runtimes/core/src/trace/protocol.rs

```diff
@@ -57,6 +57,57 @@ impl Tracer {
     }
 }
 
+pub struct LogMessageData<'a> {
+    pub source: Option<&'a Request>,
+    pub msg: &'a str,
+    pub level: LogLevel,
+    pub fields: Vec<LogField<'a>>,
+}
+
+impl Tracer {
+    #[inline]
+    pub fn log_message(&self, data: LogMessageData) -> Option<TraceEventId> {
```

Comment:
I think we don't need to return an event id. It's mainly used for log events that need to correlate "start" and "end", like beginning a database query and recording its completion.